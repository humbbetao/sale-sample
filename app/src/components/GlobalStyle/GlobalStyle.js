import { createGlobalStyle } from 'styled-components'
import { createMuiTheme } from '@material-ui/core/styles'
import breakpoints from '../../config/breakpoints'
import colors from '../Colors'
import './index.css'

export default createGlobalStyle`
  body {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
    font-family: Poppins Regular;
    font-weight: 400;
    font-size: 12px;
    line-height: 28px;
    box-sizing: border-box;
    background-color: #fff;
  }
 `

export const themeStyled = { breakpoints, colors }
export const themeMaterialUi = createMuiTheme({
  palette: {
    common: { black: '#000', white: '#fff' },
    background: { paper: '#fff', default: '#fafafa' },
    primary: {
      light: 'rgba(198, 67, 133, 1)',
      main: 'rgba(199, 69, 93, 1)',
      dark: 'rgba(146, 51, 67, 1)',
      contrastText: 'rgba(230, 230, 230, 1)',
    },
    secondary: {
      light: 'rgba(115, 235, 255, 1)',
      main: 'rgba(50, 100, 110, 1)',
      dark: 'rgba(43, 86, 95, 1)',
      contrastText: 'rgba(255, 255, 255, 1)',
    },
    error: {
      light: '#e57373',
      main: '#f44336',
      dark: '#d32f2f',
      contrastText: '#fff',
    },
    text: {
      primary: 'rgba(60, 60, 60, 1)',
      secondary: 'rgba(112, 112, 112, 1)',
      disabled: 'rgba(201, 201, 201, 1)',
      hint: 'rgba(230, 230, 230, 1)',
    },
  },
})
