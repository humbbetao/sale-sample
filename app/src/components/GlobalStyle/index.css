@font-face {
  font-family: 'Poppins Regular';
  font-style: normal;
  font-weight: 400;
  font-display: fallback;
  src: local('Poppins Regular'), local('Poppins-Regular'),
    url(../../assets/fonts/Poppins-Regular.ttf) format('truetype');
}
@font-face {
  font-family: 'Poppins Medium';
  font-style: normal;
  font-weight: 500;
  font-display: fallback;
  src: local('Poppins Medium'), local('Poppins-Medium'),
    url(../../assets/fonts/Poppins-Medium.ttf) format('truetype');
}
@font-face {
  font-family: 'Poppins Bold';
  font-style: normal;
  font-weight: 700;
  font-display: fallback;
  src: local('Poppins Bold'), local('Poppins-Bold'),
    url(../../assets/fonts/Poppins-Bold.ttf) format('truetype');
}
