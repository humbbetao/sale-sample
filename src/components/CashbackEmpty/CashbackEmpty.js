import React from 'react'
import { makeStyles } from '@material-ui/core/styles'
import Paper from '@material-ui/core/Paper'
import { Container } from '@material-ui/core'
import Typography from '@material-ui/core/Typography'
import { ReactComponent as Cofrinho } from '../../assets/cofrinho.svg'
import styled from 'styled-components'
import Button from '@material-ui/core/Button'
import Add from '@material-ui/icons/Add'

const CofrinhoIcon = styled(Cofrinho)`
  fill: ${(props) => props.theme.colors.red};
  width: 120px;
  max-width: 150px;
`

const useStyles = makeStyles((theme) => ({
  container: {
    width: '100%',
    display: 'flex',
    justifyContent: 'center',
    flexDirection: 'column',
    paddingTop: '90px',
  },
  paper: {
    display: 'flex',
    flexDirection: 'column',
    width: '100%',
    alignItems: 'center',
    height: '400px',
    justifyContent: 'space-evenly',
    padding: '16px',
    boxSizing: 'border-box',
    [theme.breakpoints.up('md')]: {
      display: 'flex',
      flexDirection: 'column',
      width: '100%',
      alignItems: 'center',
      height: '400px',
      justifyContent: 'space-evenly',
    },
  },
}))

export default function CashBackEmpty({ handleOnOpenDialog }) {
  const classes = useStyles()
  return (
    <Container classes={{ root: classes.container }} maxWidth="lg">
      <Paper classes={{ root: classes.paper }}>
        <Typography color="textPrimary" variant="h6" align="left">
          Você não tem nenhuma compra registrada
        </Typography>
        <CofrinhoIcon></CofrinhoIcon>

        <Button
          color="primary"
          component="button"
          variant="outlined"
          size="large"
          data-test="add-button"
          onClick={handleOnOpenDialog}
          startIcon={<Add />}
        >
          Nova Compra
        </Button>
      </Paper>
    </Container>
  )
}
