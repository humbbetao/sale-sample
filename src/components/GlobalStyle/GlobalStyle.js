import { createGlobalStyle } from 'styled-components'
import breakpoints from '../../config/breakpoints'
import colors from '../Colors'

export default createGlobalStyle`
  body {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
  }
 `

export const theme = { breakpoints, colors }
