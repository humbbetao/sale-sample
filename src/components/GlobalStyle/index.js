export { default, themeStyled, themeMaterialUi } from './GlobalStyle'
