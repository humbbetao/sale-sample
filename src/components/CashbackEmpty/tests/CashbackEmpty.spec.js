import React from 'react'
import { mount } from 'enzyme'
import CashbackEmpty from '../CashbackEmpty'
import { themeStyled } from '../../GlobalStyle'
import { ThemeProvider as ThemeStyled } from 'styled-components'

describe('CashbackEmpty', () => {
  it('should render correctly', () => {
    let wrapper = mount(
      <ThemeStyled theme={themeStyled}>
        <CashbackEmpty />
      </ThemeStyled>
    )

    expect(wrapper).toMatchSnapshot()
    expect(wrapper).toHaveLength(1)
  })

  it('should called handleOnOpenDialog a after click on buttons', () => {
    const handleOnOpenDialog = jest.fn()
    let wrapper = mount(
      <ThemeStyled theme={themeStyled}>
        <CashbackEmpty handleOnOpenDialog={handleOnOpenDialog} />
      </ThemeStyled>
    )
    wrapper.find('[data-test="add-button"] button').simulate('click')
    expect(handleOnOpenDialog).toHaveBeenCalled()
    expect(handleOnOpenDialog).toHaveBeenCalledTimes(1)
  })
})
