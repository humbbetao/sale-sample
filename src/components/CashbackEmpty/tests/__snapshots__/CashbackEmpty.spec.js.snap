// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CashbackEmpty should render correctly 1`] = `
<WithStyles(ForwardRef(Container))
  classes={
    Object {
      "root": "makeStyles-container-1",
    }
  }
  maxWidth="lg"
>
  <ForwardRef(Container)
    classes={
      Object {
        "disableGutters": "MuiContainer-disableGutters",
        "fixed": "MuiContainer-fixed",
        "maxWidthLg": "MuiContainer-maxWidthLg",
        "maxWidthMd": "MuiContainer-maxWidthMd",
        "maxWidthSm": "MuiContainer-maxWidthSm",
        "maxWidthXl": "MuiContainer-maxWidthXl",
        "maxWidthXs": "MuiContainer-maxWidthXs",
        "root": "MuiContainer-root makeStyles-container-1",
      }
    }
    maxWidth="lg"
  >
    <div
      className="MuiContainer-root makeStyles-container-1 MuiContainer-maxWidthLg"
    >
      <WithStyles(ForwardRef(Paper))
        classes={
          Object {
            "root": "makeStyles-paper-2",
          }
        }
      >
        <ForwardRef(Paper)
          classes={
            Object {
              "elevation0": "MuiPaper-elevation0",
              "elevation1": "MuiPaper-elevation1",
              "elevation10": "MuiPaper-elevation10",
              "elevation11": "MuiPaper-elevation11",
              "elevation12": "MuiPaper-elevation12",
              "elevation13": "MuiPaper-elevation13",
              "elevation14": "MuiPaper-elevation14",
              "elevation15": "MuiPaper-elevation15",
              "elevation16": "MuiPaper-elevation16",
              "elevation17": "MuiPaper-elevation17",
              "elevation18": "MuiPaper-elevation18",
              "elevation19": "MuiPaper-elevation19",
              "elevation2": "MuiPaper-elevation2",
              "elevation20": "MuiPaper-elevation20",
              "elevation21": "MuiPaper-elevation21",
              "elevation22": "MuiPaper-elevation22",
              "elevation23": "MuiPaper-elevation23",
              "elevation24": "MuiPaper-elevation24",
              "elevation3": "MuiPaper-elevation3",
              "elevation4": "MuiPaper-elevation4",
              "elevation5": "MuiPaper-elevation5",
              "elevation6": "MuiPaper-elevation6",
              "elevation7": "MuiPaper-elevation7",
              "elevation8": "MuiPaper-elevation8",
              "elevation9": "MuiPaper-elevation9",
              "outlined": "MuiPaper-outlined",
              "root": "MuiPaper-root makeStyles-paper-2",
              "rounded": "MuiPaper-rounded",
            }
          }
        >
          <div
            className="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <WithStyles(ForwardRef(Typography))
              align="left"
              color="textPrimary"
              variant="h6"
            >
              <ForwardRef(Typography)
                align="left"
                classes={
                  Object {
                    "alignCenter": "MuiTypography-alignCenter",
                    "alignJustify": "MuiTypography-alignJustify",
                    "alignLeft": "MuiTypography-alignLeft",
                    "alignRight": "MuiTypography-alignRight",
                    "body1": "MuiTypography-body1",
                    "body2": "MuiTypography-body2",
                    "button": "MuiTypography-button",
                    "caption": "MuiTypography-caption",
                    "colorError": "MuiTypography-colorError",
                    "colorInherit": "MuiTypography-colorInherit",
                    "colorPrimary": "MuiTypography-colorPrimary",
                    "colorSecondary": "MuiTypography-colorSecondary",
                    "colorTextPrimary": "MuiTypography-colorTextPrimary",
                    "colorTextSecondary": "MuiTypography-colorTextSecondary",
                    "displayBlock": "MuiTypography-displayBlock",
                    "displayInline": "MuiTypography-displayInline",
                    "gutterBottom": "MuiTypography-gutterBottom",
                    "h1": "MuiTypography-h1",
                    "h2": "MuiTypography-h2",
                    "h3": "MuiTypography-h3",
                    "h4": "MuiTypography-h4",
                    "h5": "MuiTypography-h5",
                    "h6": "MuiTypography-h6",
                    "noWrap": "MuiTypography-noWrap",
                    "overline": "MuiTypography-overline",
                    "paragraph": "MuiTypography-paragraph",
                    "root": "MuiTypography-root",
                    "srOnly": "MuiTypography-srOnly",
                    "subtitle1": "MuiTypography-subtitle1",
                    "subtitle2": "MuiTypography-subtitle2",
                  }
                }
                color="textPrimary"
                variant="h6"
              >
                <h6
                  className="MuiTypography-root MuiTypography-h6 MuiTypography-colorTextPrimary MuiTypography-alignLeft"
                >
                  Você não tem nenhuma compra registrada
                </h6>
              </ForwardRef(Typography)>
            </WithStyles(ForwardRef(Typography))>
            <Styled(Component)>
              <ForwardRef(SvgCofrinho)
                className="sc-AxjAm bVwRyt"
              >
                <svg
                  className="sc-AxjAm bVwRyt"
                >
                  cofrinho.svg
                </svg>
              </ForwardRef(SvgCofrinho)>
            </Styled(Component)>
            <WithStyles(ForwardRef(Button))
              color="primary"
              component="button"
              data-test="add-button"
              size="large"
              startIcon={<UNDEFINED />}
              variant="outlined"
            >
              <ForwardRef(Button)
                classes={
                  Object {
                    "colorInherit": "MuiButton-colorInherit",
                    "contained": "MuiButton-contained",
                    "containedPrimary": "MuiButton-containedPrimary",
                    "containedSecondary": "MuiButton-containedSecondary",
                    "containedSizeLarge": "MuiButton-containedSizeLarge",
                    "containedSizeSmall": "MuiButton-containedSizeSmall",
                    "disableElevation": "MuiButton-disableElevation",
                    "disabled": "Mui-disabled",
                    "endIcon": "MuiButton-endIcon",
                    "focusVisible": "Mui-focusVisible",
                    "fullWidth": "MuiButton-fullWidth",
                    "iconSizeLarge": "MuiButton-iconSizeLarge",
                    "iconSizeMedium": "MuiButton-iconSizeMedium",
                    "iconSizeSmall": "MuiButton-iconSizeSmall",
                    "label": "MuiButton-label",
                    "outlined": "MuiButton-outlined",
                    "outlinedPrimary": "MuiButton-outlinedPrimary",
                    "outlinedSecondary": "MuiButton-outlinedSecondary",
                    "outlinedSizeLarge": "MuiButton-outlinedSizeLarge",
                    "outlinedSizeSmall": "MuiButton-outlinedSizeSmall",
                    "root": "MuiButton-root",
                    "sizeLarge": "MuiButton-sizeLarge",
                    "sizeSmall": "MuiButton-sizeSmall",
                    "startIcon": "MuiButton-startIcon",
                    "text": "MuiButton-text",
                    "textPrimary": "MuiButton-textPrimary",
                    "textSecondary": "MuiButton-textSecondary",
                    "textSizeLarge": "MuiButton-textSizeLarge",
                    "textSizeSmall": "MuiButton-textSizeSmall",
                  }
                }
                color="primary"
                component="button"
                data-test="add-button"
                size="large"
                startIcon={<UNDEFINED />}
                variant="outlined"
              >
                <WithStyles(ForwardRef(ButtonBase))
                  className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-outlinedSizeLarge MuiButton-sizeLarge"
                  component="button"
                  data-test="add-button"
                  disabled={false}
                  focusRipple={true}
                  focusVisibleClassName="Mui-focusVisible"
                  type="button"
                >
                  <ForwardRef(ButtonBase)
                    className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-outlinedSizeLarge MuiButton-sizeLarge"
                    classes={
                      Object {
                        "disabled": "Mui-disabled",
                        "focusVisible": "Mui-focusVisible",
                        "root": "MuiButtonBase-root",
                      }
                    }
                    component="button"
                    data-test="add-button"
                    disabled={false}
                    focusRipple={true}
                    focusVisibleClassName="Mui-focusVisible"
                    type="button"
                  >
                    <button
                      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-outlinedSizeLarge MuiButton-sizeLarge"
                      data-test="add-button"
                      disabled={false}
                      onBlur={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={0}
                      type="button"
                    >
                      <span
                        className="MuiButton-label"
                      >
                        <span
                          className="MuiButton-startIcon MuiButton-iconSizeLarge"
                        >
                          <ForwardRef>
                            <WithStyles(ForwardRef(SvgIcon))>
                              <ForwardRef(SvgIcon)
                                classes={
                                  Object {
                                    "colorAction": "MuiSvgIcon-colorAction",
                                    "colorDisabled": "MuiSvgIcon-colorDisabled",
                                    "colorError": "MuiSvgIcon-colorError",
                                    "colorPrimary": "MuiSvgIcon-colorPrimary",
                                    "colorSecondary": "MuiSvgIcon-colorSecondary",
                                    "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                                    "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                                    "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                                    "root": "MuiSvgIcon-root",
                                  }
                                }
                              >
                                <svg
                                  aria-hidden={true}
                                  className="MuiSvgIcon-root"
                                  focusable="false"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                                  />
                                </svg>
                              </ForwardRef(SvgIcon)>
                            </WithStyles(ForwardRef(SvgIcon))>
                          </ForwardRef>
                        </span>
                        Nova Compra
                      </span>
                      <WithStyles(memo)
                        center={false}
                      >
                        <ForwardRef(TouchRipple)
                          center={false}
                          classes={
                            Object {
                              "child": "MuiTouchRipple-child",
                              "childLeaving": "MuiTouchRipple-childLeaving",
                              "childPulsate": "MuiTouchRipple-childPulsate",
                              "ripple": "MuiTouchRipple-ripple",
                              "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                              "rippleVisible": "MuiTouchRipple-rippleVisible",
                              "root": "MuiTouchRipple-root",
                            }
                          }
                        >
                          <span
                            className="MuiTouchRipple-root"
                          />
                        </ForwardRef(TouchRipple)>
                      </WithStyles(memo)>
                    </button>
                  </ForwardRef(ButtonBase)>
                </WithStyles(ForwardRef(ButtonBase))>
              </ForwardRef(Button)>
            </WithStyles(ForwardRef(Button))>
          </div>
        </ForwardRef(Paper)>
      </WithStyles(ForwardRef(Paper))>
    </div>
  </ForwardRef(Container)>
</WithStyles(ForwardRef(Container))>
`;
